### PR TITLE
Routes updated in GobiertoPlans

### LINKED ISSUE #3569: GobiertoPlans vue-router nested childs
The new tabs require another level of nesting for the vue-router in plans.

Currently, the `Home` component includes all nested components, but after including the new tabs, both `Home` (it should be renamed as *Plan*) and `Dashboards` (and another future tabs) must be at the same level.

### PR DESCRIPTION
**Pull request by Crashillo**
Closes #3569 

## :v: What does this PR do?
Handle the router UX in Plans properly:
- Updates the url accordingly the dashboards selected or not
- Load an specific dashboard if comes from url

Besides, it includes a normalization of the routes. The aim is about avoiding to deal with plain-text route names in the code, but only in one place: the router definition.

### CODE REVIEW

**Review by jorgeatgu — APPROVED** [review]

**Review by ferblape — APPROVED** [review]

File: config/routes.rb [hunk 1]
```diff
@@ -460,6 +460,7 @@
         get ":slug(/:year)/proyecto/:id" => "plan_types#show", as: :project
         get ":slug(/:year)/tabla/:uid" => "plan_types#show"
         get ":slug(/:year)/tabla/:uid/:term_id" => "plan_types#show"
+        get ":slug(/:year)/dashboards(/:dashboard_id)" => "plan_types#show"
```
Comment (ferblape) [inline config/routes.rb:463]: This change should be included in the documentation in order for the module Dashboards to work, right?

File: app/javascript/gobierto_plans/webapp/components/GroupsByTermTableRow.vue [hunk 2]
```diff
@@ -10,7 +10,7 @@
       >
         <i :class="['fas', nestedGroups.length ? `fa-caret-${isOpen ? 'down' : 'right'}` : '']" />
         <router-link
-          :to="{ name: 'term', params: { ...params, term: slug } }"
+          :to="{ name: routes.TERM, params: { ...params, term: slug } }"
```
Comment (ferblape) [inline app/javascript/gobierto_plans/webapp/components/GroupsByTermTableRow.vue:13]: I like this way of naming routes

File: app/javascript/gobierto_plans/webapp/lib/router.js [hunk 3]
```diff
@@ -22,42 +33,66 @@ export const router = new VueRouter({
...
+                },
+              ]
+            }
+          ],
         },
         {
-          path: "dashboards/:dashboardid?",
-          name: "dashboards",
-          component: Dashboards
+          path: "dashboards/:dashboardId?",
```
Comment (ferblape) [inline app/javascript/gobierto_plans/webapp/lib/router.js:91]: Maybe it'd be necessary to document the list of steps to add dashboards to an existing Gobierto module?

1 - add routes to the vue component
2 - create a tab component
3 - add rails routes
...

File: app/javascript/gobierto_plans/webapp/components/GroupsByTermTableRow.vue (hunk 2 above)
Comment (Crashillo) [reply app/javascript/gobierto_plans/webapp/components/GroupsByTermTableRow.vue:13]: Two best practices here:
1. Avoid text literals in the code
2. At the moment that one variable is reused somewhere else..... move it to shared place

File: app/javascript/gobierto_plans/webapp/lib/router.js (hunk 3 above)
Comment (Crashillo) [reply app/javascript/gobierto_plans/webapp/lib/router.js:91]: It'll depend on the module the dashboards are gonna be included. I mean, let's say a new module provides a route, `site.gobify.net/dummy/dashboards`, to render directly the dashboards. You won't need anything (but the rails routes) to make it work.
Vue-router and Tabs are specifications of the one single design, not general.

File: config/routes.rb (hunk 1 above)
Comment (Crashillo) [reply config/routes.rb:463]: You may take a look https://gobierto.readme.io/docs/gobierto-dashboards

Comment (ferblape) [reply config/routes.rb:463]: 👍🏾

**Pull request merged**